src/Language/PureScript/Ide/Matcher.hs

```diff
@@ -86,15 +86,25 @@ flexRate pattern c@(Completion (_,ident,_)) = do
 -- the matchas a (start, length) tuple if there's a match.
 -- If match fails then it would be (-1,0)
 flexScore :: Text -> DeclIdent -> Maybe Double
-flexScore "" _ = Nothing
 flexScore pat str =
-  case TE.encodeUtf8 str =~ TE.encodeUtf8 pat' :: (Int, Int) of
-    (-1,0) -> Nothing
-    (start,len) -> Just $ calcScore start (start + len)
-  where
-    Just (first,pattern) = T.uncons pat
-    -- This just interleaves the search string with .*
-    -- abcd -> a.*b.*c.*d
-    pat' = first `T.cons` T.concatMap (T.snoc ".*") pattern
-    calcScore start end =
-      100.0 / fromIntegral ((1 + start) * (end - start + 1))
+  case T.uncons pat of
+    Nothing -> Nothing
+    Just (first, pattern) ->
+      case TE.encodeUtf8 str =~ TE.encodeUtf8 pat' :: (Int, Int) of
+        (-1,0) -> Nothing
+        (start,len) -> Just $ calcScore start (start + len)
+      where
+        escapedPattern :: [Text]
+        escapedPattern = map escape (T.unpack pattern)
+
+        -- escape wraps "regexy" characters in brackets to prevent the matcher
+        -- from crashing when trying to build the regex
+        escape :: Char -> Text
+        escape c = if c `elem` ("*.?$|\\" :: String)
+                   then "[" <> T.singleton c <> "]"
```

Comment:
You need to backslash escape these, since `^`, `]`, and `\` are meaningful inside a character class.